cloudinit/templater.py

```diff
@@ -101,19 +164,30 @@ def replacer(match):
 def detect_template(text):
     def jinja_render(content, params):
         # keep_trailing_newline is in jinja2 2.7+, not 2.6
+        content = "##template:jinja\n" + content
```

Comment:
Why are we adding this cloud-init header back into the content passed to JTemplate only to remove it again upon return of JTemplate. jinja2.Template shouldn't care about this cloud-init header comments remotely. Is this to try to establish proper line posisitons??

Reply:
Yes it is to establish proper line positions upon failure. But now that I think about it, without even checking if the
header is even there in the first place, this doesn't actually add any flexibility. I'll just revert to how i had it
previously (just always adding 1 to the line number that the original jinja error returns.)